src/Trace/TraceSpan.php

```diff
@@ -0,0 +1,204 @@
+            ['spanId', 'kind', 'name', 'startTime', 'endTime', 'parentSpanId', 'labels'],
+            $options
+        );
+        $this->info += [
+            'kind' => self::SPAN_KIND_UNSPECIFIED
+        ];
+
+        if (!array_key_exists('spanId', $this->info)) {
+            $this->info['spanId'] = $this->generateSpanId();
+        }
+
+        if (!array_key_exists('name', $this->info)) {
+            $this->info['name'] = $this->generateSpanName();
+        }
+    }
+
+    /**
+     * Set the start time for this span.
+     *
+     * @param  \DateTimeInterface $when [optional] The start time of this span.
+     *         **Defaults to** now.
+     */
+    public function setStart(\DateTimeInterface $when = null)
+    {
+        $this->info['startTime'] = $this->formatDate($when);
+    }
+
+    /**
+     * Set the end time for this span.
+     *
+     * @param  \DateTimeInterface $when [optional] The end time of this span.
+     *         **Defaults to** now.
+     */
+    public function setFinish(\DateTimeInterface $when = null)
+    {
+        $this->info['endTime'] = $this->formatDate($when);
+    }
+
+    /**
+     * Retrieve the ID of this span.
+     *
+     * @return string
+     */
+    public function spanId()
+    {
+        return $this->info['spanId'];
+    }
+
+    /**
+     * Retrieve the name of this span.
+     *
+     * @return string
+     */
+    public function name()
+    {
+        return $this->info['name'];
+    }
+
+    /**
+     * Returns a serializable array representing this span.
+     *
+     * @return array
+     */
+    public function info()
+    {
+        return $this->info;
+    }
+
+    /**
+     * Attach labels to this span.
+     *
+     * @param array $labels Labels in the form of $label => $value
+     */
+    public function addLabels(array $labels)
+    {
+        foreach ($labels as $label => $value) {
+            $this->addLabel($label, $value);
+        }
+    }
+
+    /**
+     * Attach a single label to this span.
+     *
+     * @param string $label The name of the label.
+     * @param mixed $value The value of the label. Will be cast to a string
+     */
+    public function addLabel($label, $value)
+    {
+        if (!array_key_exists('labels', $this->info)) {
+            $this->info['labels'] = [];
+        }
+        $this->info['labels'][$label] = (string) $value;
+    }
+
+    /**
+     * Returns a "Zulu" formatted string representing the
+     * provided \DateTime.
+     *
+     * @param  \DateTimeInterface $when [optional] The end time of this span.
+     *         **Defaults to** now.
+     * @return string
+     */
+    private function formatDate(\DateTimeInterface $when = null)
+    {
+        if (!$when) {
+            list($usec, $sec) = explode(' ', microtime());
+            $micro = sprintf("%06d", $usec * 1000000);
+            $when = new \DateTime(date('Y-m-d H:i:s.' . $micro));
+        }
+        return $when->format('Y-m-d\TH:i:s.u000\Z');
```

Comment:
I think we need to make sure the timezone is set to UTC here, or the system default will be used. 

```php
$when->setTimezone(new \DateTimeZone('UTC'));
```